src/render/unified/shader.wgsl

```diff
@@ -77,14 +96,18 @@ fn fragment(in: VertexOutput) -> @location(0) vec4<f32> {
         return vec4<f32>(in.color.rgb, rect_dist * in.color.a);
     }
     if quad_type.t == 1 {
-        var px_range = 5.0;
+        var px_range = 5.;
         var tex_dimensions = textureDimensions(font_texture);
-        var msdf_unit = vec2<f32>(px_range, px_range) / vec2<f32>(f32(tex_dimensions.x), f32(tex_dimensions.y));
-        var x = textureSample(font_texture, font_sampler, vec2<f32>(in.uv.x, 1.0 - in.uv.y), i32(in.uv.z));
-        var v = max(min(x.r, x.g), min(max(x.r, x.g), x.b));
-        var sig_dist = (v - 0.5) * dot(msdf_unit, 0.5 / fwidth(in.uv.xy));
-        var a = clamp(sig_dist + 0.5, 0.0, 1.0);
-        return vec4<f32>(in.color.rgb, a);
+        var msdf_unit = vec2(px_range, px_range) / vec2(f32(tex_dimensions.x), f32(tex_dimensions.y));
+        let scale = dot(msdf_unit, 0.5 / fwidth(in.uv.xy));
+        let subpixel_dimensions = vec2(fwidth(in.uv.x) / 3., fwidth(in.uv.y));
+
+        let red = sample_subpixel(vec2(in.uv.x - subpixel_dimensions.x, 1. - in.uv.y), subpixel_dimensions, i32(in.uv.z), scale);
+        let green = sample_subpixel(vec2(in.uv.x, 1. - in.uv.y), subpixel_dimensions, i32(in.uv.z), scale);
+        let blue = sample_subpixel(vec2(in.uv.x + subpixel_dimensions.y, 1. - in.uv.y), subpixel_dimensions, i32(in.uv.z), scale);

```

Comment:
👀 `+ subpixel_dimensions.y`

Reply:
![image](https://user-images.githubusercontent.com/17050131/204107245-53c180ad-61e2-4412-9375-ff121e0312bb.png)

still a hack, but fixing this bug improves the colour a lot 😅